src/lib/transition_frontier_persistence/intf.ml

```diff
@@ -104,6 +114,36 @@ module type Main_inputs = sig
     type t
 
     val handle_diff :
-      t -> Diff_hash.t -> 'a Diff_mutant.t -> Diff_hash.t Deferred.Or_error.t
+         t
+      -> Diff_hash.t
+      -> State_hash.t Diff_mutant.E.t
+      -> Diff_hash.t Deferred.Or_error.t
+  end
+end
+
+module type S = sig
+  type frontier
+
+  type worker
+
+  type 'output diff
+
+  type diff_hash
+
+  val listen_to_frontier_broadcast_pipe :
+       logger:Logger.t
+    -> frontier option Broadcast_pipe.Reader.t
+    -> worker
+    -> unit Deferred.t
+
+  (* TODO: Lol this cannot be polymorphic. Don't actually get any wins if this was polymorphic *)
```

Comment:
are you talking about the `'output` type? is this TODO done?

Reply:
yep, I deleted it and is unnecessary.